cli/src/cli_util.rs

```diff
@@ -876,6 +876,9 @@ impl WorkspaceCommandEnvironment {
         &self,
         ui: &Ui,
     ) -> Result<Arc<UserRevsetExpression>, CommandError> {
+        if self.command.global_args().ignore_immutable {
```

Comment:
iirc, the idea was that immutable revisions are still rendered as such even if `--ignore-immutable` is specified.